### PR TITLE
Added mysqlpath option

### PR DESCRIPTION
**Pull request by sbktrifork**
##### ISSUE TYPE
- Feature Pull Request
- Bugfix Pull Request
##### COMPONENT NAME

database/mysql/mysql_db.py 
##### ANSIBLE VERSION

2.0.1.0
##### SUMMARY

Adds a mysqlpath option, so one can specify the path to the mysql binaries mysql and mysqldump. 
I didn't use module.get_path, since there might be multiple mysql binaries in the path. 

It also fixes transaction and quick option for mysqldump aka db_dump. 
quick and transaction parameters were not parsed to the db_dump method before. 

Before:

mysql_db: "state=import login_unix_socket=/tmp/mysql.sock login_user={{ sql_database_user }} login_password={{ sql_database_password }} name={{ sql_database_schema }} target={{ sql_database_temp }}/{{ sql_database_schema }}/{{ item.item }}.sql"

resulted in 

failed: [clienthost -> dbhost] => (item={'_ansible_no_log': False, 'item': u'whitelist', '_ansible_delegated_vars': {'ansible_host': u'rn-db03.nspnet.dk'}, u'failed': True, 'invocation': {'module_name': u'mysql_query', u'module_args': {u'login_port': 3306, u'login_user': u'root', u'login_host': u'localhost', u'db': u'sts', u'login_unix_socket': u'/tmp/mysql.sock', u'login_password': u'xxxxxx', u'query': u'show create table mytable'}}, u'msg': u'select failed due to "(1146, "Table \'db.mytable\' doesn\'t exist")".'}) => {"failed": true, "item": {"_ansible_delegated_vars": {"ansible_host": "dbhost"}, "_ansible_no_log": false, "failed": true, "invocation": {"module_args": {"db": "mydb", "login_host": "localhost", "login_password": "xxxxxxx", "login_port": 3306, "login_unix_socket": "/tmp/mysql.sock", "login_user": "root", "query": "show create table mytable"}, "module_name": "mysql_query"}, "item": "mytable", "msg": "select failed due to \"(1146, \"Table 'mydb.mytable' doesn't exist\")\"."}, "msg": "Failed to find required executable mysql"}

After:

mysql_db: "state=import login_unix_socket=/tmp/mysql.sock login_user={{ sql_database_user }} login_password={{ sql_database_password }} name={{ sql_database_schema }} mysqlpath=/pack/mysql/bin target={{ sql_database_temp }}/{{ sql_database_schema }}/{{ item.item }}.sql"

... it works fine...

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @sorenk. To the current maintainers, @Jmainguy please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
There should be a much simpler way by using `environment` http://docs.ansible.com/ansible/playbooks_environment.html

```
mysql_db: 
  state=import
  login_unix_socket=/tmp/mysql.sock
  login_user={{ sql_database_user }}
  login_password={{ sql_database_password }}
  name={{ sql_database_schema }}
  target={{ sql_database_temp }}/{{ sql_database_schema }}/{{ item.item }}.sql
environment: 
  PATH: "/pack/mysql/bin"
```

Could you test if this would work?

needs_info

**Comment by sbktrifork** [comment]
Absolutely - adding environment works perfect. I wasn't aware of the feature - sorry about that.

File: database/mysql/mysql_db.py [hunk 1]
```diff
@@ -313,7 +324,7 @@ def main():
             else:
                 rc, stdout, stderr = db_dump(module, login_host, login_user,
                                             login_password, db, target, all_databases,
-                                            login_port, config_file, socket, ssl_cert, ssl_key, ssl_ca)
+                                            login_port, config_file, socket, ssl_cert, ssl_key, ssl_ca, single_transaction, quick, mysqlpath)
```
Comment (sbktrifork) [inline database/mysql/mysql_db.py:327]: This one still needs to be fixed though.. (without the mysqlpath argument of course)
Should I do a new pull request on this?

**Comment by resmo** [comment]
yes please

**Comment by gregdek** [comment]
@sorenk, a community member requested some information about this PR; for that reason, we put this PR into the 'needs_info' state.

When you have answered the questions, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by sbktrifork** [comment]
OK, I will create a new pull request...

**Pull request closed without merging**